### DISCUSSION TITLE
Multiples connection factories with unequal isPublisherConfirms in RabbitTemplate

### DISCUSSION THREAD

**Discussion by LeonardoFerreiraa** [spring-projects/spring-amqp#1431] [General]
Hello guys, I have a case here that I want to use 2 connection factories in the same `RabbitTemplate`. One `CachingConnectionFactory` with `ConfirmType = Correlated` and another with type `PooledCachingConnectionFactory`. I need this because in some cases I want to be sure of the message delivery and in others, I don't need this.

I found in the `RabbitTemplate` the property `sendConnectionFactorySelectorExpression` that allows me to have more than one `ConnectionFactory` based on a SPEL, but for this, I need an instance of `AbstractRoutingConnectionFactory` that has a validation avoiding different `isPublisherConfirms` and `isPublisherReturns` values between the factories.

If `RabbitTemplate` didn't depend on `AbstractRoutingConnectionFactory` but `RoutingConnectionFactory`, and some more small changes in `confirmsOrReturnsCapable` property, it would be possible;

What do you think about it?

**Comment by garyrussell** [comment]
This is probably historical.

The preferred way for to obtain a confirmation is (now) via the `Future<?>` in the `CorrelationData`; previously it had to be via a callback passed into the template.

For backwards compatibility, I suppose the check should be performed if the user provides callbacks; we could also consider deprecating the callbacks.

I have no objections otherwise; I will create an issue from this.

**Reply by garyrussell** [reply]
https://github.com/spring-projects/spring-amqp/issues/1434

**Reply by LeonardoFerreiraa** [reply]
Nice. Do you see any problem if I work on it and open a PR?

**Reply by garyrussell** [reply]
That would be great; thanks.
